cyences_app_for_splunk/default/data/ui/views/cs_network_reports.xml

```diff
@@ -57,56 +74,339 @@
+| `drop_dm_object_name(All_Traffic)` 
+| append 
+    [| inputlookup cs_tenable_vuln 
+    | search (vul_state="open" OR vul_state="reopened") vul_severity IN ($tkn_vul_severity$) 
+    | fields tenable_uuid 
+    | join tenable_uuid type=left 
+        [| inputlookup cs_tenable_inventory 
+        | table tenable_uuid, ip] 
+    | makemv ip delim="~~" 
+    | stats count by ip 
+    | rename ip as src_ip 
+    | eval is_vul = 1 ] 
+| append 
+    [| search `cs_qualys_vuln` `cs_qualys_timerange` STATUS="ACTIVE" severity IN ($tkn_vul_severity$) 
+    | fields IP 
+    | stats count by IP 
+    | rename IP as src_ip 
+    | eval is_vul = 1 ] 
+| eval single_field = dest_ip."#".dest_port."#".action."#".total_bytes."#".total_packets  
+| stats count, values(single_field) as single_field, values(is_vul) as is_vul by src_ip 
+| search single_field="*" 
+| mvexpand single_field 
+| rex field=single_field "^(?&lt;dest_ip&gt;[^#]+)#(?&lt;dest_port&gt;[^#]+)#(?&lt;action&gt;[^#]+)#(?&lt;total_bytes&gt;[^#]+)#(?&lt;total_packets&gt;[^#]+)$" 
+| fillnull is_vul 
+| iplocation dest_ip 
+| rename lat as end_lat, lon as end_lon, Country as dest_country, City as dest_city, Region as dest_region 
+| eval start_lat=`cs_network_home_location_lat`, start_lon=`cs_network_home_location_lon`, src_country=`cs_home_country` 
+| sort 0 -is_vul, -action, -total_bytes 
+| eval total_KB = round(total_bytes/1024, 2) 
+| table src_ip, src_country, dest_ip, dest_port, dest_country, dest_city, dest_region, action, is_vul, total_KB, total_packets, start_lat, start_lon, end_lat, end_lon</option>
+        <option name="cyences_app_for_splunk.network_telemetry_map.mapLatitude">30</option>
+        <option name="cyences_app_for_splunk.network_telemetry_map.mapLongitude">0</option>
+        <option name="cyences_app_for_splunk.network_telemetry_map.mapZoom">2</option>
+        <option name="drilldown">none</option>
+        <option name="refresh.display">progressbar</option>
+      </viz>
+      <html>
+      <style>
+      table, th, td {
+      	border: 1px solid; 
+        text-align: center;
+    
+      }
+      th, tr {
+          padding: 15px
+      }
+      </style>
+        <div>
+          <p>
+            <b>Description:</b> Displays a traffic visual for the top twenty networks for each action and is_vulnerable combination with their respective source and destination pathways.</p>
+        </div>
         <div>
           <p>
-            <b>Description:</b> Displays a traffic visual for the top twenty networks with their respective source and destination pathways.</p>
+            <b>Traffic Color Mapping</b>
```

Comment:
How to read the chart?

| Color | Description |
| -------- | ------------ |
| Red | Outbound traffic on the **vulnerable devices** that are **allowed** by the firewall. |
| Orange | Outbound traffic on the **non-vulnerable devices** that are **blocked** by the firewall. |
| Blue | Other Outbound traffic is **blocked** by the firewall. (Only top 20) |
| Green | Other Outbound traffic **allowed** by the firewall. (Only top 20) |

Vulnerable Device - If there is at least one open vulnerability with severity selected by the user in the dashboard filter.